### DISCUSSION TITLE
VRP - Vehicle Breaks not working properly

### DISCUSSION THREAD

**Discussion by alvesdotcom** [google/or-tools#2956] [Routing (and legacy CP) questions]
I'm trying to implement Vehicles breaks without success.

I have implemented a simple case were only one order and one vehicle exists.
There are no Time Windows defined, so the order can be delivered from time 0 to kHorizon (48h).
The travel time from depot to the order is 40k seconds (11.1h).
There are two mandatory vehicle breaks:

- The first one starts at 1h and has a duration of 6h
- The second one starts at 12h and has a duration of 3h

Considering this data, the vehicle should be able to arrive at the order around 20.1h as follows:
1h (T) + 6h (B) + 5h (T) + 3h (B) + 5.1h (T)
where T = Travel Time and B = Break Time

But the order is dropped by the routing solver.
If one of the breaks is removed the order is delivered correctly.

I created the following gist with this sample: https://gist.github.com/alvesdotcom/fda7074daf6e9540c3aa05828586adff

Can someone confirm if this issue is in OrTools or in my gist?

Thank you!

**Comment by 270ajay** [comment]
Try using this `MakeFixedDurationIntervalVar`. Using this worked for me in python. Maybe breaks method has bugs, see https://github.com/google/or-tools/issues/569

src: https://github.com/google/or-tools/blob/master/examples/cpp/cvrptw_with_breaks.cc

**Comment by GuyBenhaim** [comment]
Hello,
Is this working, at least with one break per vehicle? more than one break?

I'm using C#, is this limited to py? How do we define:  vector<IntervalVar*>?
Any example implementations are welcome.

Otherwise, must we use all sorts of tricks to support breaks?
Guy

**Comment by alvesdotcom** [comment]
@270ajay thanks for the hint but I get similar results when using `MakeFixedDurationIntervalVar`. With one or two small breaks between the Depot and the Order, the Order is always dropped.

In one of my tests I inserted three non optional breaks before the Order could be delivered. The assignment says that the breaks were performed and the Order delivered but the delivered time is the same as the one without breaks!

I don't know if this is a bug or not, I'm just trying to figure it out.

@GuyBenhaim with just one break I get some logic results, but with multiple it starts to give weird results...

**Comment by arvsr1988** [comment]
@alvesdotcom have you tried to increase the penalty for dropping a node to a high amount? That might help avoid the jobs being dropped?

**Comment by alvesdotcom** [comment]
@arvsr1988 unfortunately, increasing the penalties doesn't have an effect in the results

**Comment by mathiascbaur** [comment]
@Mizux 
Are there any news on this topic for v7.9? Being able to use `SetBreakIntervalsOfVehicle` in our use cases would be extremely helpful.
